### PR TITLE
feat(extension-api): Allows to register a factory for kubernetes connection objects (like for container connections)

### PR DESCRIPTION
**Pull request by benoitf**
### What does this PR do?
Add missing items for .kubernetesConnections object of providers

part of https://github.com/containers/podman-desktop/issues/413

### Screenshot/screencast of this PR

N/A

### What issues does this PR fix or reference?
related to https://github.com/containers/podman-desktop/issues/413

### How to test this PR?
it should just work as before for current providers.
Check that the resource page for container providers is not broken as now one page is shared between container and kubernetes connections

Change-Id: Ic85b01c20253062899b67cc72616fc4d4ecbccd7
Signed-off-by: Florent Benoit <fbenoit@redhat.com>

### CODE REVIEW

File: packages/renderer/src/lib/preferences/PreferencesKubernetesConnectionRendering.svelte [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+
+let showModal: ProviderInfo = undefined;
+
+let logsTerminal;
+
+async function startReceivinLogs(provider: ProviderInfo): Promise<void> {
+  const logHandler = (newContent: any[], colorPrefix: string) => {
+    writeToTerminal(logsTerminal, newContent, colorPrefix);
+  };
+  /*window.startReceiveLogs(
```
Comment (cdrage) [inline packages/renderer/src/lib/preferences/PreferencesKubernetesConnectionRendering.svelte]: forgot to remove code?

**Review by cdrage — CHANGES_REQUESTED** [review]
super small nitpick otherwise LGTM

**Comment by cdrage** [comment]
Confirm that resource page is not broken! Amazing code @benoitf 🎉

File: packages/renderer/src/lib/preferences/PreferencesKubernetesConnectionRendering.svelte (hunk 1 above)
Comment (benoitf) [reply packages/renderer/src/lib/preferences/PreferencesKubernetesConnectionRendering.svelte]: fixed

**Review by cdrage — APPROVED** [review]

**Pull request merged**